src/main/java/org/jetbrains/research/refactorinsight/processors/CommitMiner.java

```diff
@@ -65,20 +65,24 @@ public CommitMiner(ExecutorService pool, Map<String, RefactoringEntry> map,
   /**
    * Method that mines only one commit.
    *
-   * @param commit  commit metadata
-   * @param map     the inner map that should be updated
-   * @param project the current project
-   * @param repository Git Repository
+   * @param commitHash       commit hash.
+   * @param commitParentHash commit parent's hash.
+   * @param commitTimestamp  commit timestamp.
+   * @param map              the inner map that should be updated.
+   * @param project          the current project.
+   * @param repository       Git Repository.
    */
-  public static void mineAtCommit(TimedVcsCommit commit, Map<String, RefactoringEntry> map,
+  public static void mineAtCommit(String commitHash, String commitParentHash, long commitTimestamp,
```

Comment:
We only pass one parent commit as a reference point. 
Can we handle merge commits that have more than one parent and a non-trivial diff against all parents?

Reply:
If I'm not mistaken, we **store** parent commit hash only for the refactoring history feature, we don't use it in any other way. RefactoringMiner needs only a repository and a commit hash to detect the refactorings.